elasticgraph-warehouse/spec/unit/elastic_graph/warehouse/schema_definition/scalar_type_extension_spec.rb

```diff
@@ -24,83 +22,35 @@ module SchemaDefinition
         end
 
         it "handles built-in scalar types" do
-          results = define_warehouse_schema do |s|
-            s.json_schema_version 1
-          end
-
-          # Ensure on_built_in_types callbacks are executed before accessing scalar types
-          results.send(:all_types)
+          results = define_warehouse_schema
 
           # Test the scalar types directly
-          expect(results.state.scalar_types_by_name["Int"].to_warehouse_column_type).to eq("INT")
-          expect(results.state.scalar_types_by_name["Float"].to_warehouse_column_type).to eq("DOUBLE")
-          expect(results.state.scalar_types_by_name["Boolean"].to_warehouse_column_type).to eq("BOOLEAN")
-          expect(results.state.scalar_types_by_name["String"].to_warehouse_column_type).to eq("STRING")
-          expect(results.state.scalar_types_by_name["ID"].to_warehouse_column_type).to eq("STRING")
-        end
-
-        it "converts custom scalar type with warehouse_column to warehouse column type" do
-          results = define_warehouse_schema do |s|
-            s.json_schema_version 1
-
-            s.scalar_type "CustomType" do |t|
-              t.mapping type: "keyword"
-              t.json_schema type: "string"
-              t.warehouse_column type: "BINARY"
-            end
-          end
-
-          # Ensure on_built_in_types callbacks are executed before accessing scalar types
-          results.send(:all_types)
-
-          scalar_type = results.state.scalar_types_by_name["CustomType"]
-          expect(scalar_type.to_warehouse_column_type).to eq("BINARY")
+          expect(warehouse_column_type_for(results, "Int")).to eq("INT")
+          expect(warehouse_column_type_for(results, "Float")).to eq("DOUBLE")
+          expect(warehouse_column_type_for(results, "Boolean")).to eq("BOOLEAN")
+          expect(warehouse_column_type_for(results, "String")).to eq("STRING")
+          expect(warehouse_column_type_for(results, "ID")).to eq("STRING")
         end
 
         it "raises an error when a custom scalar type does not configure warehouse_column" do
           results = define_warehouse_schema do |s|
-            s.json_schema_version 1
-
             s.scalar_type "UnconfiguredScalar" do |t|
               t.mapping type: "keyword"
               t.json_schema type: "string"
               # Intentionally NOT calling t.warehouse_column
             end
           end
 
-          # Ensure on_built_in_types callbacks are executed before accessing scalar types
-          results.send(:all_types)
-
-          scalar_type = results.state.scalar_types_by_name["UnconfiguredScalar"]
-
           expect {
-            scalar_type.to_warehouse_column_type
-          }.to raise_error(RuntimeError, /Warehouse column type not configured for scalar type "UnconfiguredScalar".*Call `warehouse_column type:/)
+            warehouse_column_type_for(results, "UnconfiguredScalar")
+          }.to raise_error(Errors::SchemaError, a_string_including(
```

Comment:
Using regex is nice but `a_string_including` is a bit simpler when you don't need the full power of regex (and want characters like `.` to be treated literally).